spacepy/irbempy/irbempy.py

```diff
@@ -1433,16 +1435,29 @@ def _get_Lstar(ticks, loci, alpha, extMag='T01STORM', options=[1,0,0,0,0], omniv
     nTAI = len(ticks)
     d = prep_irbem(ticks, loci, alpha, extMag, options, omnivals)
     nalpha = d['nalpha']
+    if d['kext'] != 5 or d['options'][4] != 0:
+        landi2lstar = False
+    no_shell_splitting = (nalpha == 0) or (nalpha == 1 and alpha[0] == 90)
         
-    if isinstance(alpha, numbers.Number): # no drift shell splitting
-        raise Exception
-        lm, lstar, blocal, bmin, xj, mlt = oplib.make_lstar1(nTAI, d['kext'], d['options'], d['sysaxes'],\
-                    d['iyearsat'], d['idoysat'], d['utsat'], d['xin1'], d['xin2'], d['xin3'], d['magin'])
+    if no_shell_splitting: # no drift shell splitting
+        if landi2lstar:
+            lm, lstar, blocal, bmin, xj, mlt = oplib.landi2lstar1(nTAI, d['kext'], d['options'], \
+                d['sysaxes'],d['iyearsat'], d['idoysat'], d['utsat'], d['xin1'], d['xin2'], \
+                d['xin3'], d['magin'])
+        else:
+            lm, lstar, blocal, bmin, xj, mlt = oplib.make_lstar1(nTAI, d['kext'], d['options'], \
+                d['sysaxes'],d['iyearsat'], d['idoysat'], d['utsat'], d['xin1'], d['xin2'], \
+                d['xin3'], d['magin'])
 
     elif nalpha > 0 and nalpha <= d['nalp_max']: # with drift shell splitting
-        lm, lstar, bmirr, bmin, xj, mlt = oplib.make_lstar_shell_splitting1(nTAI, nalpha, \
-            d['kext'], d['options'], d['sysaxes'], d['iyearsat'], d['idoysat'], d['utsat'], \
-            d['xin1'], d['xin2'], d['xin3'], d['degalpha'], d['magin'])
+        if landi2lstar:
```

Comment:
I wonder if it might be clearer that the only difference between these is the function called by doing something like:
```
func = oplib.landi2lstar_shell_splitting1 if landi2lstar else oplib.make_lstar_shell_splitting
lm, lstar, bmirr, bmin, xj, mlt = func(all the args)
```
Something similar might collapse this whole if tree down...I can whip up something if that sounds reasonable. It's just easy to miss the differences when there's so much similar.

Reply:
Yes, I believe it would be clearer.